src/migration/KnexMigrationSource.ts

```diff
@@ -47,15 +49,16 @@ class KnexMigrationSource {
   /**
    * Get the migration runner.
    *
-   * @param {SqlMigrationEntry} migration
+   * @param {string} name
+   * @returns {Promise<MigrationRunner>}
    */
-  getMigration(name: string) {
+  getMigration(name: string): Promise<MigrationRunner> {
     const migration = this.migrationContext.get(name);
 
     this.log(`getMigration - ${name}`);
     this.log(`getMigration - resolve: %o`, migration);
 
-    return migration;
+    return Promise.resolve(migration);
```

Comment:
Do we need to return a Promise? knex documentation just returns the migration context here. 

https://knexjs.org/guide/migrations.html#custom-migration-sources

Reply:
Yes, this part bothered me as well. The type definition says it returns a Promise though. 🤯 . 
![migrationtype](https://user-images.githubusercontent.com/45115406/234832425-4acbfffd-8ea6-4644-a9d8-467c87619f89.jpg)

Reply:
Understood.